src/cocotb/_scheduler.py

```diff
@@ -424,6 +422,11 @@ def _schedule_task(
         for tasks in self._trigger2tasks.values():
             if task in tasks:
                 return
```

Comment:
These checks right here are rather expensive on simulations of any size. What's worse is it doing these checks every time an existing Task is rescheduled after a Trigger fires. This check isn't needed in those cases. So we only do the check on user interfaces.